config/routes.rb

```diff
@@ -2814,19 +2810,15 @@ def create_api_route(verb, url, action)
                          "/api(/:version)/#{collection_name}(/*c_suffix)",
                          action_for_collection(collection_name, verb))
       else
-        if collection.options.include?(:subcollection) || collection.subcollections
-          create_api_route(verb, "/api(/:version)/#{collection_name}(/:c_id)", action_for(verb))
-        else
-          create_api_route(verb, "/api(/:version)/#{collection_name}(/:c_id)", action_for_collection(collection_name, verb))
-        end
+        create_api_route(verb, "/api(/:version)/#{collection_name}(/:c_id)", action_for_collection(collection_name, verb))
       end
     end
 
     Array(collection.subcollections).each do |subcollection_name|
       Api::Settings.collections[subcollection_name].verbs.each do |verb|
         create_api_route(verb,
                          "/api(/:version)/#{collection_name}/:c_id/#{subcollection_name}(/:s_id)",
-                         action_for(verb))
+                         action_for_collection(collection_name, verb))
```

Comment:
Not crazy about this. I'd really like for a subcollection to route to its own controller (not that of the parent collection as it is here) and so avoid having to include all that subcollection code into the parent controllers too. But there's some dependencies that need to be carefully teased apart, and it was too big a job to do at once. Hopefully this is satisfactory as a stepping stone